src/app/snapp_test_transaction/snapp_test_transaction.ml

```diff
@@ -0,0 +1,1247 @@
+      totalCurrency: null, 
+      minWindowDensity: null, 
+      blockchainLength: null, 
+      timestamp: null, 
+      snarkedNextAvailableToken: null, 
+      snarkedLedgerHash: null}
+    |}
+  in
+  let party (p : Party.t) =
+    let authorization = authorization p.authorization in
+    let predicate =
+      match p.data.predicate with
+      | Nonce n ->
+          sprintf "{ nonce: \"%s\" }" (Account.Nonce.to_string n)
+      | Full _a ->
+          (*TODO*)
+          {|
+          {
+            account: {
+              balance:null,
+              nonce:null,
+              receiptChainHash:null,
+              publicKey:null,
+              delegate:null,
+              state:
+                {elements: [null,null,null,null,null,null,null,null]},
+              sequenceState:null,
+              provedState:null}
+          }
+          |}
+      | Accept ->
+          "{account:null, nonce:null}"
+    in
+    let balance_change =
+      let sgn =
+        match Currency.Amount.Signed.sgn p.data.body.balance_change with
+        | Pos ->
+            "PLUS"
+        | Neg ->
+            "MINUS"
+      in
+      let magnitude =
+        Currency.Amount.(
+          to_string (Signed.magnitude p.data.body.balance_change))
+      in
+      sprintf "{sign: %s, magnitude: \"%s\"}" sgn magnitude
+    in
+    let increment_nonce = Bool.to_string p.data.body.increment_nonce in
+    let sequence_events =
+      `List
+        (List.map p.data.body.sequence_events ~f:(fun a ->
+             `List
+               ( Array.map a ~f:(fun s ->
+                     `String (Snark_params.Tick.Field.to_string s))
+               |> Array.to_list )))
+      |> Yojson.Safe.to_string
+    in
+    let call_data =
+      `String (Snark_params.Tick.Field.to_string p.data.body.call_data)
+      |> Yojson.Safe.to_string
+    in
+    let events =
+      `List
+        (List.map p.data.body.events ~f:(fun a ->
+             `List
+               ( Array.map a ~f:(fun s ->
+                     `String (Snark_params.Tick.Field.to_string s))
+               |> Array.to_list )))
+      |> Yojson.Safe.to_string
+    in
+    let full_commitment = Bool.to_string p.data.body.use_full_commitment in
+    let pk = pk_string p.data.body.pk in
+    sprintf
+      {|
+        authorization: %s, 
+      data: {
+        predicate: %s, 
+        body: {
+          callDepth: 0, 
+          callData: %s, 
+          sequenceEvents: %s, 
+          events: %s, 
+          balance_change: %s, 
+          tokenId: "1",
+          incrementNonce: %s 
+          update: {
+            timing: null, 
+            tokenSymbol: null, 
+            snappUri: null, 
+            permissions: %s, 
+            verificationKey: %s, 
+            delegate: null, 
+            appState: [%s]}, 
+          publicKey: "%s",
+          protocolState: %s,
+          use_full_commitment: %s}}
+    |}
+      authorization predicate call_data sequence_events events balance_change
```

Comment:
This long list of stringly-typed substitutions feels quite brittle. However, it's not immediately clear to me what a better alternative would be..

Reply:
@mrmr1993 If I recall correctly, the JSON is only a little different than what GraphQL expects (like the `PLUS` and `MINUS`. Could we instead retrieve the JSON, patch it, and use `Yojson.Safe.to_string`?

Maybe that could be a subsequent PR.